hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -922,22 +936,25 @@ int QuectelNcpClient::muxChannelStateCb(uint8_t channel, decltype(muxer_)::Chann
     // This callback is executed from the multiplexer thread, not safe to use the lock here
     // because it might get called while blocked inside some muxer function
 
+    // Also please note that connectionState() should never be called with the CONNECTED state
+    // from this callback.
+
     // We are only interested in Closed state
     if (newState == decltype(muxer_)::ChannelState::Closed) {
         switch (channel) {
-        case 0: {
-            // Muxer stopped
-            self->disable();
-            self->connState_ = NcpConnectionState::DISCONNECTED;
-            break;
-        }
-        case QUECTEL_NCP_PPP_CHANNEL: {
-            // PPP channel closed
-            if (self->connState_ != NcpConnectionState::DISCONNECTED) {
-                self->connState_ = NcpConnectionState::CONNECTING;
+            case 0: {
+                // Muxer stopped
+                self->disable();
```

Comment:
Will not work in this order. I would argue that `connectionState()` call is not needed here.